language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/RegexpCompletionProvider.java

```diff
@@ -0,0 +1,688 @@
+                    RegexpCompletionProvider.RegexSnippet.DEF_QUANTIFIER_ASTERISK_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_QUANTIFIER_PLUS_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_QUANTIFIER_QUESTION_MARK_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_LEFT_BRACE_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_RIGHT_BRACE_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_LEFT_SQUARE_BRACE_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_RIGHT_SQUARE_BRACE_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_LEFT_CURLY_BRACE_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_RIGHT_CURLY_BRACE_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_PIPE_CHAR.get()));
+
+            // unicode property chars
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_UNICODE_PROPERTY_CHAR.snippetBlock));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_NON_UNICODE_PROPERTY_CHAR.snippetBlock));
+
+        } else if (isFlagExpr(nodeAtCursor, ctx)) {
+            // Eg: re `(?<cursor>)`
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_MULTILINE_FLAG.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_DOT_ALL_FLAG.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_IGNORE_CASE_FLAG.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_COMMENT_FLAG.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexSnippet.DEF_COLON_CHAR.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexSnippet.DEF_MINUS_CHAR.get()));
+            return completionItems;
+        } else if (nodeAtCursor.kind() == SyntaxKind.RE_QUOTE_ESCAPE && nodeAtCursor.toSourceCode().equals("\\p")) {
```

Comment:
Again, prefer if we can use syntax kinds and token at cursor to achieve this instead of string comparisions.

Reply:
No we cannot.